### ISSUE TITLE
Prepare works for sharding-scaling contribute to ShardingSphere

### ISSUE THREAD

**Issue by KomachiSion** [apache/incubator-shardingsphere#3115]
There are some based function and prepare works need to be done before sharding-scaling contributing to ShardingSphere.

- [x] Add Apache License
- [x] Remove or replace illegal dependences
    - [x] Remove mysql-connector-java jar
    - [x] Remove mysql-binlog-connector-java jar [sharding-scaling#3](https://github.com/avalon566/sharding-scaling/issues/3)
- [x] MySQL support
    - [x] Connect Channel to MySQL
    - [x] Parse binlog event  [sharding-scaling#3](https://github.com/avalon566/sharding-scaling/issues/3)
- [x] ~~PostgreSQL support~~ (implement after merge)
- [x] ~~Integrate with ShardingSphere~~ (implement after merge)
    - [x] ~~Change to a Service process~~ (implement after merge)
    - [x] ~~Provide API to generate or stop migrate jobs~~ (implement after merge)
- [x] CheckStyle
- [x] Operation document

If some other works need to be done or some works can be done after contributing, please discuss in comment

reference: 
1.  #3077 
2. [sharding-scaling#2](https://github.com/avalon566/sharding-scaling/issues/2)

**Comment by SanmuWangZJU** [comment]
how do u deal with real-time data while migrating history data？

**Comment by KomachiSion** [comment]
> how do u deal with real-time data while migrating history data？

By parsing binlog or wal.

**Comment by avalon566** [comment]
@KomachiSion @terrymanu I think we can discuss about Which is import job for merge into ss.

**Comment by KomachiSion** [comment]
I think the last job `Integrate with ShardingSphere` can be done after merged.

**Comment by avalon566** [comment]
PostgreSQL support also can be done after merged. I think is ready to merge.

**Comment by KomachiSion** [comment]
There are two more things we should prepare before merge
1. Code checkstyle. If the checkstyle can't pass, it will install failed after merge.
2. Documents. we should add some documents, at least add operation document to teach users how to use this component. After merged, we can migrate these document to [apache/incubator-shardingsphere-doc](https://github.com/apache/incubator-shardingsphere-doc)

**Comment by ssxlulu** [comment]
A document about the overview of the component may be added in the prepare works list I think, people will know the architecture、design idea and processing flow about the component.
Is anyone writing it？

**Comment by avalon566** [comment]
@ssxlulu You can create an issue and assign it to yourself.

**Comment by ssxlulu** [comment]
> @ssxlulu You can create an issue and assign it to yourself.

https://github.com/avalon566/sharding-scaling/issues/15 , I have opened this issue.

**Comment by ssxlulu** [comment]
Have anyone done a performance test？

**Comment by avalon566** [comment]
@ssxlulu No.

**Comment by terrymanu** [comment]
Good job. This issue has finished.
